namenode/namenode.go

```diff
@@ -68,3 +75,69 @@ func (server *NameNode) GetAttributes(path string, fileAttributes *common.FileAt
 	*fileAttributes = server.inodes[path].fileAttributes
 	return nil
 }
+
+func (server *NameNode) RegisterDataNode(dataNodeEndpoint string, success *bool) error {
+	slog.Info("RegisterDataNode request", "dataNodeEndpoint", dataNodeEndpoint)
+
+	server.globalRWLock.Lock()
+	defer server.globalRWLock.Unlock()
+
+	// TODO: Check if the datanode is already registered
+	server.datanodes[dataNodeEndpoint] = DataNodeInfo{
+		LastTimestamp: time.Now(),
+		isAlive:       true,
+	}
+
+	*success = true
+	return nil
+}
+
+func (server *NameNode) ReceiveBlockReport(blockReport common.BlockReport, success *bool) error {
+	slog.Info("ReceiveBlockReport request", "dataNodeEndpoint", blockReport.Endpoint)
+
+	server.globalRWLock.Lock()
+	defer server.globalRWLock.Unlock()
+
+	// TODO: handle block report
+
+	*success = true
+	return nil
+}
+
+func (server *NameNode) ReceiveHeartBeat(heartbeat common.Heartbeat, success *bool) error {
+	slog.Info("ReceiveHeartBeat request", "dataNodeEndpoint", heartbeat.Endpoint)
+
+	server.globalRWLock.Lock()
+	defer server.globalRWLock.Unlock()
+
+	if _, exists := server.datanodes[heartbeat.Endpoint]; !exists {
+		*success = false
+		slog.Error("ReceiveHeartBeat datanode not registered", "dataNodeEndpoint", heartbeat.Endpoint)
+		return errors.New("datanode not registered")
+	}
+
+	// Update the last timestamp and status of the datanode
+	server.datanodes[heartbeat.Endpoint] = DataNodeInfo{
+		LastTimestamp: time.Now(),
+		isAlive:       true,
+	}
+
+	*success = true
+	return nil
+}
+
+func (server *NameNode) HeartbeatMonitor() {
```

Comment:
I guess this method doesn't need to be exported. If so, it had better start with a lower-case letter.